tests/toranj/build.sh

```diff
@@ -120,7 +121,7 @@ case ${build_config} in
         cd "${top_builddir}" || die "cd failed"
         cmake -GNinja -DOT_PLATFORM=simulation -DOT_COMPILE_WARNING_AS_ERROR=ON -DOT_COVERAGE=${ot_coverage} \
             -DOT_THREAD_VERSION=1.3.1 -DOT_APP_CLI=OFF -DOT_APP_NCP=ON -DOT_APP_RCP=OFF \
-            -DOT_15_4=ON -DOT_TREL=OFF \
+            -DOT_15_4=ON -DOT_TREL=OFF -DOT_OPERATIONAL_DATASET_AUTO_INIT=ON \
             -DOT_CONFIG=../tests/toranj/openthread-core-toranj-config-simulation.h \
```

Comment:
I think we use this config on all builds, so may be better to add it in `openthread-core-toranj-config-simulation.h` as:
```
#define OPENTHREAD_CONFIG_OPERATIONAL_DATASET_AUTO_INIT 1
```